docs/architecture/adr-008-blocksize-independent-commitment.md

```diff
@@ -0,0 +1,100 @@
+[`HashFromByteSlices`](https://github.com/celestiaorg/celestia-core/blob/v0.34.x-celestia/crypto/merkle/tree.go#L7-L21) results now in the merkleroot `Com'`, which is not equal to `Com`.
+
+![Old Commitment](./assets/size-dependent-commitment.png)
+
+To have a commitment independent of the block size, you would need to create a merkle tree over subtree roots that are included in every possible block size.
+We agree on producing the commitment as if the message was put into the `minimalBlocksize`. Therefore the subtree roots will be the same and the resulting commitment as well, no matter the block size. *This is because all the subtree roots in the `minimalBlocksize` are also included in bigger block sizes*.
+You can see this in the updated version of the diagram above. Now the merkle root will stay the same. This property results from the block width and length being a power of 2.  
+
+![New Commitment](./assets/size-independent-commitment.png)
+
+You can see in the diagramm below that no matter how the message is split up into rows, the hashes of the subtree roots stay the same.
+
+![Row Size Comparison](./assets/row-size-comparison.png)
+
+This scheme works for interactive commitments as well as long as the index of when a message starts is the same in every block size. In other words, the leaf nodes and leaf node pairs are independent of the block size.
+For example, in the following diagram, you see the message starting at the second share and still having the same commitment. I marked the skipped blocks pink to show that both messages have the same starting index.
+
+![Interactive Commitment](./assets/interactive-commitment.png)
+
+If the message starting point index is larger than the row of `minimalBlocksize`, you take the index mod `minimalBlocksize`. The `minimalBlocksize` is shown in green, and the skipped block is in pink. In both cases, the commitment stays the same.
+
+![Interactive Commitment 2](./assets/interactive-commitment2.png)
+
+## Alternative Approaches
+
+TODO
+
+## Decision
+
+TODO
+
+## Detailed Design
+
+To implement this decision, you need to change [`CreateCommit`](https://github.com/celestiaorg/celestia-app/blob/0c81704939cd743937aac2859f3cb5ae6368f174/x/payment/types/payfordata.go#L112-166).
+In Detail, [`powerOf2MountainRange`](https://github.com/celestiaorg/celestia-app/blob/0c81704939cd743937aac2859f3cb5ae6368f174/x/payment/types/payfordata.go#L142) should take `minimalBlocksize` as an argument instead of `squareSize`.
+
+Following the non-interactive default rules, `minimalBlocksize` can be calculated like this:
+
+```go
+func minimalBlocksize(messageLength uint64) uint64 {
+    blocksize := NextHigherPowerOf2(messageLength)
+    //Check if message fits with non-interactive default rules
+    if messageLength <= (blocksize * (blocksize -1)) {
+        return blocksize
+    } else {
+        return blocksize << 1
+    }
+}
+```
+
+## Status
+
+Proposed
```

Comment:
imo, making a decision on the status of this should not be a blocker for merging. We should merge it regardless.

Reply:
@musalbas